### DISCUSSION TITLE
07-02-2021 Latest update and can no longer login.

### DISCUSSION THREAD

**Discussion by DDD-Digithead** [Security-Onion-Solutions/securityonion#4682] [General]
OK, so I ran soup this morning as part of my morning routine. It took a while to complete but apparently did so successfully. However, now I can no longer log into the web interface. It just keeps cycling back to the login page.

Suggestions? Am I the only one having this problem?

**Comment by dougburks** [comment]
Does the web interface show any errors?

Have you tried shift-reload in your browser?

Have you tried resetting your password?

Have you tried rebooting?

Have you checked the logs for additional clues?

**Reply by DDD-Digithead** [reply]
Yes for all of the above save for the web interface shows no errors and last one. No idea what log to look for. Have also tried different browsers with the same result.

**Reply by jertel** [reply]
Could be something wrong in Kratos. Does the so-kratos container show as OK in `so-status`? The log file for Kratos is located at `/opt/so/log/kratos/kratos.log`.

**Reply by DDD-Digithead** [reply]
I just rebooted it again so it's still on the process of starting everything back up but everything was running just fine according to so-status.

**Reply by DDD-Digithead** [reply]
Another update to it this morning and still no joy. I added another user just to try and get into SOC but that doesn't work either. Entering user name and password just cycles right back to the login, no errors... nothing.

Do I need to just reload it from scratch again?

**Reply by DDD-Digithead** [reply]
I think I've stumbled onto what's going on. I revisited "/opt/so/log/kratos/kratos.log" and found this which I believe to be the issue:

{"audience":"audit","error":{"debug":"","message":"request does not have a valid authentication session","reason":"No active session was found in this request.","status":"Unauthorized","status_code":401},"http_request":{"headers":{"accept":"*/*","accept-encoding":"gzip,identity","accept-language":"en-us,en;q=0.5","referer":"http://eyeofsauron/","user-agent":"WWW-Mechanize/1.0.0 (http://rubyforge.org/projects/mechanize/)","x-forwarded-for":"192.168.0.244","x-forwarded-proto":"https"},"host":"eyeofsauron","method":"GET","path":"/sessions/whoami","query":null,"remote":"172.17.0.1:54072","scheme":"http"},"level":"info","msg":"No valid session cookie found.","service_name":"Ory Kratos","service_version":"master","time":"2021-07-03T03:18:01Z"}

Searching further for similar I found this:

https://github.com/ory/kratos/issues/97

After clearing cookies on all of my browsers I was able to log in again but only with Firefox. Chrome still won't work though. So at least I'm getting to the bottom of it.

**Reply by DDD-Digithead** [reply]
So... still cannot get it to work... Chrome won't connect. Guess it's just time to reload it yet again...

**Reply by jertel** [reply]
The Kratos log snippet doesn't look like a problem - that is normal behavior for when SOC is asking if a user is already logged in. If not, it will return that 401 which then tells SOC to show the login form. Look for something more along the lines of a CSRF error, password failure, or anything that would be associated with an HTTP post to a login flow.

**Comment by Stephen-Ridgway** [comment]
We're getting this error (or something very similar):

![image](https://user-images.githubusercontent.com/80473697/124291618-a5c3c480-db4c-11eb-8d40-1eaf15ca8ce9.png)

**Reply by Stephen-Ridgway** [reply]
soc won't start because of this error

**Reply by Stephen-Ridgway** [reply]
![image](https://user-images.githubusercontent.com/80473697/124291946-0226e400-db4d-11eb-904d-e5ee2c9cd454.png)

**Reply by DDD-Digithead** [reply]
Not seeing any errors in /opt/so/log/kratos/kratos.log
Things are still starting up right now but I did check that log.

**Reply by dougburks** [reply]
@Stephen-Ridgway To avoid confusion, we'd prefer that you start a new discussion rather than reply to somebody else's discussion. Thanks!

**Reply by Stephen-Ridgway** [reply]
No probs. Just looked like the same thing.

**Comment by LWHaysUSDC** [comment]
I had similar behavior on a new install of 2.3.52  [(link)](https://github.com/Security-Onion-Solutions/securityonion/discussions/4718) and it turned out to be browser-related - Chrome/Chromium didn't login (just recycled as you describe) but Firefox worked.   I'm running these browsers on Ubuntu linux desktops (specifically mint and POP!OS)

Hope this helps.

L

**Reply by DDD-Digithead** [reply]
@jertel @LWHaysUSDC Sure enough... that seems to be exactly what's happening. Edge, Internet Exploader and Chrome... none of those will log in from any machine on the LAN. Only Firefox will work. Oh well... as long I can log into it that'll do. Really wouldn't bother me to just reload from scratch as it really isn't that difficult to do. But I will NOT remove this device from the network as long as I can get into it. On my network, it's sort of like taking a shot-gun to a butterfly hunt but Security Onion is by far the best tool I've messed with and in the 30 years in this field I've tried a lot  of them.

**Reply by jertel** [reply]
Thank you both for the feedback on this issue. I've not run into this problem. But if it does show up for me I'll certainly flush out the bug, whatever it is. In the meantime if anyone can reproduce this and wants to export the network logs (HAR format via DevTools -> Network tab) so I can inspect the requests/responses, that could also give me a clue to what's going on.

**Reply by LWHaysUSDC** [reply]
@jertel It occurred to me that both failing browsers have the lastpass extention installed.  I have other extensions, but that's the one in common and I wonder if @DDD-Digithead also has lastpass installed in his problematic browsers?    I'll try to get the network logs for you once my current work trip is over.

**Reply by DDD-Digithead** [reply]
Hi @LWHaysUSDC , no... I don't use LastPass so no extension for it in any of the browsers. I do have two extensions that I suspected (Ublock Origin and Malwarebytes) might be causing it but I disabled them both for testing purposes and same issue. For the time being I've just been using Firefox to log into SOC.

**Reply by LWHaysUSDC** [reply]
@jertel I removed the lastpass extension and still got the same behavior.   I have generated network logs and will send them to you by PM, if possible, to (over) protect  my employer's confidentiality.

**Reply by LWHaysUSDC** [reply]
@jertel here are the .har files (hopefully helpful), encrypted with your public key
[SObrowserDebug.zip](https://github.com/Security-Onion-Solutions/securityonion/files/6804357/SObrowserDebug.zip)

**Reply by DDD-Digithead** [reply]
Keeping an eye open here to see how this goes.

**Reply by jertel** [reply]
@DDD-Digithead Please review this post to see if your situation is similar: https://github.com/Security-Onion-Solutions/securityonion/discussions/4718#discussioncomment-1000545

**Reply by DDD-Digithead** [reply]
It looks as though this is indeed what the issue is. In my case, the quickest and easiest thing to do would be to reload from scratch as soon as I have time to do so. Not a big deal for me really.

**Reply by DDD-Digithead** [reply]
I give up on this. That last update from last week broke it again. Could not get to anything aside from 'Alerts' and 'Grafani'. Went to reload this morning and latest ISO won't even boot now. I just don't have the time that the 'care and feeding' this seems to require to keep it running.
